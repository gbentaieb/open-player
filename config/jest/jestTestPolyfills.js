// make fullscreen functions available
document.requestFullscreen = () => {};
document.exitFullscreen = () => {};
document.fullscreenElement = () => {};
document.fullscreenEnabled = () => {};
document.fullscreenchange = () => {};
document.fullscreenerror = () => {};
