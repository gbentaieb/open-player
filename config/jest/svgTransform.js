// This is a custom Jest transformer turning svg imports into empty objects.

module.exports = {
  process() {
    return 'module.exports = {};';
  },
  getCacheKey() {
    // The output is always the same.
    return 'svgTransform';
  },
};
