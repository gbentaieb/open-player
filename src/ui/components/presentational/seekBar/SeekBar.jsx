import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Slider from '@material-ui/lab/Slider';
import { Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';

const styles = {
  root: {
    display: 'flex',
    width: '100%',
  },
  sliderContainer: {
    display: 'flex',
    width: '100%',
    margin: 'auto',
    padding: '0px 5px 0px 5px',
  },
  timeLabel: {
    margin: 'auto',
    padding: '0px 5px 0px 5px',
    color: 'white',
    fontSize: '0.7rem',
  },
  slider: {
    padding: '10px 0px 10px 0px',
  },
  sliderTrackAfter: {
    backgroundColor: 'white',
    opacity: 0.5,
  },
};

class SeekBar extends Component {
  static propTypes = {
    currentTime: PropTypes.number,
    startTime: PropTypes.number,
    endTime: PropTypes.number,
    seekTo: PropTypes.func.isRequired,
    classes: PropTypes.object.isRequired,
  }

  static defaultProps = {
    currentTime: 0,
    startTime: 0,
    endTime: Infinity,
  }

  static convertTime(seconds) {
    const hours = Math.floor(seconds / 3600);
    const mins = Math.floor((seconds / 60) % 60);
    const secs = Math.floor(seconds % 60);

    const stdHours = hours > 0 && hours < 10
      ? `0${hours}`
      : hours;
    const stdMins = mins < 10
      ? `0${mins}`
      : mins;
    const stdSecs = secs < 10
      ? `0${secs}`
      : secs;

    const hoursStr = stdHours ? `${stdHours}:` : '';
    const minsStr = `${stdMins}:`;
    const secsStr = stdSecs;

    return `${hoursStr}${minsStr}${secsStr}`;
  }

  constructor(props) {
    super(props);
    this.handleSliderValueChange = this.handleSliderValueChange.bind(this);
  }

  getSliderValue() {
    const { currentTime, startTime, endTime } = this.props;
    return ((currentTime - startTime) / (endTime - startTime)) * 100 || 0;
  }

  handleSliderValueChange(event, value) {
    const { startTime, endTime } = this.props;
    const seekTime = (endTime - startTime) * (value / 100);
    this.props.seekTo(seekTime);
  }

  render() {
    const { classes, currentTime, endTime } = this.props;

    return (
      <div className={classes.root}>
        <div className={classes.timeLabel}>
          <Typography variant="inherit" color="inherit">
            { SeekBar.convertTime(currentTime) }
          </Typography>
        </div>
        <div className={classes.sliderContainer}>
          <Slider
            classes={{ container: classes.slider, trackAfter: classes.sliderTrackAfter }}
            value={this.getSliderValue()}
            aria-labelledby="label"
            onChange={this.handleSliderValueChange}
          />
        </div>
        <div className={classes.timeLabel}>
          <Typography variant="inherit" color="inherit">
            {`-${SeekBar.convertTime(endTime - currentTime)}`}
          </Typography>
        </div>
      </div>
    );
  }
}

export { SeekBar };
export default withStyles(styles)(SeekBar);
