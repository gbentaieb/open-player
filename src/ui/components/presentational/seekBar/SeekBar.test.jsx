import React from 'react';
import jest from 'jest-mock';
import { SeekBar } from './SeekBar';

const defaultProps = {
  seekTo: () => {},
  classes: {},
  currentTime: 1,
  startTime: 0,
  endTime: 2,
};

describe('SeekBar snapshot', () => {
  test('SeekBar > Snapshot', () => {
    const wrapper = shallow(<SeekBar {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('SeekBar > getSliderValue', () => {
    const wrapper = shallow(<SeekBar {...defaultProps} />);
    const result = wrapper.instance().getSliderValue();
    expect(result).toEqual(50);
  });

  test('SeekBar > handleSliderValueChange', () => {
    const seekTo = jest.fn();
    const wrapper = shallow(<SeekBar {...defaultProps} seekTo={seekTo} />);
    wrapper.instance().handleSliderValueChange(null, 50);
    expect(seekTo).toHaveBeenCalledWith(1);
  });
});
