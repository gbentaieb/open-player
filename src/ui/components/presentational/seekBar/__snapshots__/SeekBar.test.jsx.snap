// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeekBar snapshot SeekBar > Snapshot 1`] = `
<div>
  <WithStyles(Slider)
    aria-labelledby="label"
    classes={
      Object {
        "container": undefined,
        "trackAfter": undefined,
      }
    }
    onChange={[Function]}
    value={50}
  />
</div>
`;
